datalad/runner/runner.py

```diff
@@ -63,14 +63,33 @@ def __init__(self,
     def _get_adjusted_env(self,
                           env: dict | None = None,
                           cwd: str | PathLike | None = None,
-                          copy: bool = True):
-        """Return an adjusted copy of an execution environment
+                          copy: bool = True
+                          ) -> dict | None:
+        """Return an adjusted execution environment
 
-        Or return an unaltered copy of the environment, if no adjustments
-        need to be made.
+        This method adjusts the environment provided in `env` to
+        reflect the configuration of the runner. It returns
+        an altered copy or an altered original, if `copy` is
+        `False`.
+
+        Parameters
+        ----------
+        env: dict
```

Comment:
I think we should stop adding type annotations in docstrings since it duplicates type hints in the signature and can quickly (like immediately in this case) diverge: signature allows for `None`.

Filed related https://github.com/datalad/datalad/issues/7404, so may be just 

```suggestion
        env
```
? but if you decide to keep type annotations then I don't think they are correct since see other uses in the codebase and https://numpydoc.readthedocs.io/en/latest/format.html  and below AFAIK should be

```
        cwd: str or PathLike, optional
```

Reply:
and now we use those hints, so best probably to just remove type annotations in the docstrings.